src/common/MessageDispatcher.sol

```diff
@@ -110,6 +112,42 @@ contract MessageDispatcher is Auth, IMessageDispatcher {
         }
     }
 
+    /// @inheritdoc IPoolMessageSender
+    function sendNotifySharePrice(uint16 chainId, PoolId poolId, ShareClassId scId, D18 sharePrice) external auth {
+        uint64 timestamp = block.timestamp.toUint64();
+        if (chainId == localCentrifugeId) {
+            poolManager.updateSharePrice(poolId.raw(), scId.raw(), sharePrice.raw(), timestamp);
+        } else {
+            gateway.send(
+                chainId,
+                MessageLib.NotifySharePrice({
+                    poolId: poolId.raw(),
+                    scId: scId.raw(),
+                    price: sharePrice.raw(),
+                    timestamp: timestamp
+                }).serialize()
+            );
+        }
+    }
+
+    function sendNotifyAssetPrice(PoolId poolId, ShareClassId scId, AssetId assetId, D18 assetPrice) external auth {
+        uint64 timestamp = block.timestamp.toUint64();
+        if (assetId.chainId() == localCentrifugeId) {
+            poolManager.updateAssetPrice(poolId.raw(), scId.raw(), assetId.raw(), assetPrice.raw(), timestamp);
```

Comment:
@mustermeiszer  what's the product reason behind notifying the asset price to CV, where CV can already ask for the price itself, given the asset lives in CV?

Reply:
All calcaulations on the CV side must be aware about CP. Not really a realistic thing